cmd/mdatagen/internal/metric.go

```diff
@@ -101,7 +101,7 @@ func (m *Metric) validate() error {
 	return errs
 }
 
-func (m *Metric) Unmarshal(parser *confmap.Conf) error {
+func (m Metric) Unmarshal(parser *confmap.Conf) error {
```

Comment:
I think this should fail because you parse in a copy.

Reply:
Oops. Guess I'll backtrack and go by pointer for everything then.